// @eui/cli/lib/skeletons/_angular/module-ngrx/src/lib/src/components/component1/component1.component.ts
import { Component, OnInit, Inject } from '@angular/core';
import { Store } from '@ngrx/store';
import { Observable } from 'rxjs/Observable';
import { CONFIG_TOKEN } from '@eui/core';

import { namespace } from '../../constants';
import { AddEntityAction, LoadDataAction, ClearEntitiesAction } from '../../actions/index';
import { getSelectedEntity } from '../../reducers/index';

@Component({
    selector: '@module.name@-component1',
    templateUrl: './component1.component.html',
    styleUrls: ['./component1.component.scss'],
})
export class Component1Component implements OnInit {

    selectedEntity: Observable<any>;

    constructor(@Inject(CONFIG_TOKEN) private config: any, private store: Store<any>) {}

    ngOnInit() {
        console.log(this.config, namespace);
        this.selectedEntity = this.store.select(getSelectedEntity);
    }

    addEntity() {
        const id = Math.floor(Math.random() * 1000);
        const entities = { id, title: `My title ${id}` };
        this.store.dispatch(new AddEntityAction(entities));
    }

    loadData() {
        this.store.dispatch(new LoadDataAction());
    }

    clearEntities() {
        this.store.dispatch(new ClearEntitiesAction());
    }
}
