// @eui/cli/lib/cli-module.js
const inquirer = require('inquirer');

function start(callback) {
    getDefaultInputs(function () {
        callback({
            "name": arguments[0].name,
            "scope": arguments[0].scope,
            "ngrx": arguments[0].ngrx
        });
    });
}

function getDefaultInputs(callback) {
    var questions = [
        {
            name: 'name',
            type: 'input',
            message: 'Enter the module name : kebab case !!!',
            default: 'my-module',
            validate: function (value) {
                if (value.length) {
                    return true;
                }
            },
        },
        {
            name: 'scope',
            type: 'input',
            message: 'Enter the module scope :',
            default: '@app',
            validate: function (value) {
                if (value.length) {
                    return true;
                }
            }
        },
        {
            name: 'ngrx',
            type: 'confirm',
            message: 'Module implements ngRx Redux store/actions/reducers ?',
            default: true
        }
    ];

    inquirer.prompt(questions).then(callback);
}

module.exports.start = start;
