// @eui/cli/lib/config.js
const path = require('path');

const skeletonsPath = path.join(__dirname, 'skeletons');
const version = require(path.join(__dirname, '../package.json')).version;

module.exports.targetPath = path.resolve(process.cwd());
module.exports.skeletonsPath = skeletonsPath;
module.exports.angularCommonsPath = path.join(skeletonsPath, '_angular/_commons');
module.exports.angularCommonsRootPath = path.join(skeletonsPath, '_angular/_commons-root');
module.exports.angularDefaultPath = path.join(skeletonsPath, '_angular/default');
module.exports.angularExtendedPath = path.join(skeletonsPath, '_angular/extended');
module.exports.webToolsPath = path.join(skeletonsPath, '_commons/web-tools');

module.exports.webMavenPath = path.join(skeletonsPath, 'web-maven');
module.exports.fullStackPath = path.join(skeletonsPath, 'full-stack');
module.exports.htmlStylesPath = path.join(skeletonsPath, 'html-styles');
module.exports.webSpringBootPath = path.join(skeletonsPath, 'web-spring-boot');

module.exports.moduleBaseSrcPath = path.join(skeletonsPath, '_angular/module-base');
module.exports.modulePath = path.join(skeletonsPath, '_angular/module');
module.exports.moduleNgRxPath = path.join(skeletonsPath, '_angular/module-ngrx');

module.exports.optionsPath = {
    "openid-jwt": path.join(skeletonsPath, '_angular/options/openid-jwt')
};

module.exports.version = version;
