// @eui/tools/prev/src/config/karma.conf.pkg.js
function get(config, package, inputOpt) {

    const DEFAULT_OPTS = {
        files: [],
        basePath: '../../..',
        packageRootPath: 'src/packages/',
        shimRootPath: null
    };

    const opt = Object.assign(DEFAULT_OPTS, inputOpt);

    var shimRootPath = opt.packageRootPath + package;

    if (opt.shimRootPath) {
        shimRootPath = opt.shimRootPath;
    }

    // transpiled app JS and map files
    const libPath = opt.packageRootPath + package + '/out-tsc/lib-es2015/';

    const FILES = [
        // System.js for module loading
        'node_modules/systemjs/dist/system.src.js',

        // Polyfills
        'node_modules/core-js/client/shim.js',
        'node_modules/systemjs/dist/system-polyfills.js',

        // zone.js
        'node_modules/zone.js/dist/zone.js',
        'node_modules/zone.js/dist/long-stack-trace-zone.js',
        'node_modules/zone.js/dist/proxy.js',
        'node_modules/zone.js/dist/sync-test.js',
        'node_modules/zone.js/dist/jasmine-patch.js',
        'node_modules/zone.js/dist/async-test.js',
        'node_modules/zone.js/dist/fake-async-test.js',

        // RxJs
        { pattern: 'node_modules/rxjs/**/*.js', included: false, watched: false },
        { pattern: 'node_modules/rxjs/**/*.js.map', included: false, watched: false },

        { pattern: 'node_modules/@ngrx/**/*.js', included: false, watched: false },
        { pattern: 'node_modules/@ngx-translate/core/**/*.js', included: false, watched: false },
        { pattern: 'node_modules/@eui/**/*.js', included: false, watched: false },
        { pattern: 'node_modules/angular2-moment/**/*.js', included: false, watched: false },
        { pattern: 'node_modules/cleave.js/**/*.js', included: false, watched: false },
        { pattern: 'node_modules/extend/index.js', included: false, watched: false },
        { pattern: 'node_modules/moment/**/*.js', included: false, watched: false },
        { pattern: 'node_modules/moment-timezone/**/*.js', included: false, watched: false },
        { pattern: 'node_modules/lodash/**/*.js', included: false, watched: false },
        { pattern: 'node_modules/lodash.template/**/*.js', included: false, watched: false },
        { pattern: 'node_modules/primeng/**/*.js', included: false, watched: false },
        { pattern: 'node_modules/tslib/**/*.js', included: false, watched: false },
        { pattern: 'node_modules/reselect/**/*.js', included: false, watched: false },

        // Paths loaded via module imports:
        // Angular itself
        { pattern: 'node_modules/@angular/**/*.js', included: false, watched: false },
        { pattern: 'node_modules/@angular/**/*.js.map', included: false, watched: false },

        shimRootPath + '/karma-test-shim.js', // optionally extend SystemJS mapping e.g., with barrels

        // transpiled application & spec code paths loaded via module imports
        { pattern: libPath + '**/*.js', included: false, watched: true },

        // Paths for debugging with source maps in dev tools
        // { pattern: libPath + '**/*.ts', included: false, watched: false },
        { pattern: libPath + '**/*.js.map', included: false, watched: false }
    ];

    const defaultConfig = {
        basePath: opt.basePath,
        frameworks: ['jasmine'],

        plugins: [
            require('karma-jasmine'),
            require('karma-chrome-launcher'),
            // require('karma-phantomjs-launcher'),
            require('karma-jasmine-html-reporter'),
            require('karma-html-reporter'),
            require('karma-coverage-istanbul-reporter'),
            require('karma-coverage'),
            require('karma-sourcemap-loader'),
        ],

        client: {
            builtPaths: [libPath], // add more spec base paths as needed
            clearContext: false // leave Jasmine Spec Runner output visible in browser
        },

        files: FILES.concat(opt.files),

        // Proxied base paths for loading assets
        proxies: {
            // required for modules fetched by SystemJS
            'node_modules/': '/base/node_modules/'
        },

        exclude: [],
        preprocessors: {},
        reporters: ['progress', 'coverage', 'kjhtml', 'html'],

        htmlReporter: {
            outputDir: opt.packageRootPath + package + '/test/reports/', // where to put the reports
            templatePath: null, // set if you moved jasmine_template.html
            focusOnFailures: true, // reports show failures on start
            namedFiles: true, // name files instead of creating sub-directories
            pageTitle: null, // page title for reports; browser info by default
            urlFriendlyName: false, // simply replaces spaces with _ for files/dirs
            reportName: 'test-report-summary', // report summary filename; browser info by default

            // experimental
            preserveDescribeNesting: false, // folded suites stay folded
            foldAll: false, // reports start folded (only with preserveDescribeNesting)
        },

        coverageReporter: {
            dir: opt.packageRootPath + package + '/test/reports/',
            includeAllSources: true,
            reporters: [
                {type: 'json', subdir: 'coverage', file: 'coverage-final.json'},
                {type: 'lcovonly', subdir: 'coverage-lcov'},
            ]
        },

        customLaunchers: {
            MyHeadlessChrome: {
                base: 'ChromeHeadless',
                flags: [
                    '--no-sandbox',
                    '--remote-debugging-port=9223',
                    '--disable-translate',
                    '--disable-extensions'
                ]
            }
        },

        port: 9876,
        colors: true,
        logLevel: config.LOG_INFO,
        autoWatch: true,
        // browsers: ['PhantomJS'],
        browsers: ['MyHeadlessChrome'],
        singleRun: false,
        browserDisconnectTolerance: 2,
        browserNoActivityTimeout: 50000
    };

    defaultConfig.preprocessors[opt.packageRootPath + package + '/out-tsc/lib-es2015/src/**/!(*.spec).js'] = 'coverage';
    defaultConfig.preprocessors[opt.packageRootPath + package + '/out-tsc/lib-es2015/src/**/*.js'] = 'sourcemap';

    return defaultConfig;
}

module.exports.get = get;
