// @netbasal/content-loader/esm5/netbasal-content-loader.js
import { __extends } from 'tslib';
import { Component, Input, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

function uid() {
    return Math.random()
        .toString(36)
        .substring(2);
}
var ContentLoaderComponent = /** @class */ (function () {
    function ContentLoaderComponent() {
        this.animate = true;
        this.width = 400;
        this.height = 130;
        this.speed = 2;
        this.preserveAspectRatio = 'xMidYMid meet';
        this.primaryColor = '#f9f9f9';
        this.secondaryColor = '#ecebeb';
        this.primaryOpacity = 1;
        this.secondaryOpacity = 1;
        this.idClip = uid();
        this.idGradient = uid();
    }
    return ContentLoaderComponent;
}());
ContentLoaderComponent.decorators = [
    { type: Component, args: [{
                selector: 'content-loader',
                template: "<svg attr.viewBox=\"0 0 {{width}} {{height}}\" version=\"1.1\" [attr.preserveAspectRatio]=\"preserveAspectRatio\">\n\n  <rect attr.clip-path=\"url(#{{ idClip }})\" [ngStyle]=\"{fill: 'url(#' + idGradient + ')' }\" x=\"0\" y=\"0\" [attr.width]=\"width\"\n    [attr.height]=\"height\" />\n\n  <defs>\n\n    <clipPath [attr.id]=\"idClip\">\n      <ng-content></ng-content>\n    </clipPath>\n\n    <linearGradient [attr.id]=\"idGradient\">\n\n      <stop offset=\"0%\" [attr.stop-color]=\"primaryColor\" [attr.stop-opacity]=\"primaryOpacity\">\n        <ng-container *ngIf=\"animate\">\n          <animate attributeName=\"offset\" values=\"-2; 1\" attr.dur={{speed}}s repeatCount=\"indefinite\" />\n        </ng-container>\n      </stop>\n\n      <stop offset=\"50%\" [attr.stop-color]=\"secondaryColor\" [attr.stop-opacity]=\"secondaryOpacity\">\n        <ng-container *ngIf=\"animate\">\n          <animate attributeName=\"offset\" values=\"-1.5; 1.5\" attr.dur={{speed}}s repeatCount=\"indefinite\" />\n        </ng-container>\n      </stop>\n\n      <stop offset=\"100%\" [attr.stop-color]=\"primaryColor\" [attr.stop-opacity]=\"primaryOpacity\">\n        <ng-container *ngIf=\"animate\">\n          <animate attributeName=\"offset\" values=\"-1; 2\" attr.dur={{speed}}s repeatCount=\"indefinite\" />\n        </ng-container>\n      </stop>\n\n    </linearGradient>\n\n  </defs>\n\n</svg>\n"
            },] },
];
ContentLoaderComponent.propDecorators = {
    "animate": [{ type: Input },],
    "width": [{ type: Input },],
    "height": [{ type: Input },],
    "speed": [{ type: Input },],
    "preserveAspectRatio": [{ type: Input },],
    "primaryColor": [{ type: Input },],
    "secondaryColor": [{ type: Input },],
    "primaryOpacity": [{ type: Input },],
    "secondaryOpacity": [{ type: Input },],
    "uniqueKey": [{ type: Input },],
};
var BulletListContentLoaderComponent = /** @class */ (function (_super) {
    __extends(BulletListContentLoaderComponent, _super);
    function BulletListContentLoaderComponent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return BulletListContentLoaderComponent;
}(ContentLoaderComponent));
BulletListContentLoaderComponent.decorators = [
    { type: Component, args: [{
                selector: 'bullet-list-content-loader',
                template: "\n  <content-loader [animate]=\"animate\" [width]=\"width\" [height]=\"height\" [speed]=\"speed\" [preserveAspectRatio]=\"preserveAspectRatio\"\n  [primaryColor]=\"primaryColor\" [secondaryColor]=\"secondaryColor\" [primaryOpacity]=\"primaryOpacity\" [secondaryColor]=\"secondaryColor\"\n  [uniqueKey]=\"uniqueKey\">\n    <svg:circle cx=\"10\" cy=\"20\" r=\"8\" />\n    <svg:rect x=\"25\" y=\"15\" rx=\"5\" ry=\"5\" width=\"220\" height=\"10\" />\n    <svg:circle cx=\"10\" cy=\"50\" r=\"8\" />\n    <svg:rect x=\"25\" y=\"45\" rx=\"5\" ry=\"5\" width=\"220\" height=\"10\" />\n    <svg:circle cx=\"10\" cy=\"80\" r=\"8\" />\n    <svg:rect x=\"25\" y=\"75\" rx=\"5\" ry=\"5\" width=\"220\" height=\"10\" />\n    <svg:circle cx=\"10\" cy=\"110\" r=\"8\" />\n    <svg:rect x=\"25\" y=\"105\" rx=\"5\" ry=\"5\" width=\"220\" height=\"10\" />\n   </content-loader>\n  "
            },] },
];
var FacebookContentLoaderComponent = /** @class */ (function (_super) {
    __extends(FacebookContentLoaderComponent, _super);
    function FacebookContentLoaderComponent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return FacebookContentLoaderComponent;
}(ContentLoaderComponent));
FacebookContentLoaderComponent.decorators = [
    { type: Component, args: [{
                selector: 'facebook-content-loader',
                template: "\n  <content-loader [animate]=\"animate\" [width]=\"width\" [height]=\"height\" [speed]=\"speed\" [preserveAspectRatio]=\"preserveAspectRatio\"\n  [primaryColor]=\"primaryColor\" [secondaryColor]=\"secondaryColor\" [primaryOpacity]=\"primaryOpacity\" [secondaryColor]=\"secondaryColor\"\n  [uniqueKey]=\"uniqueKey\">\n  \t<svg:rect x=\"70\" y=\"15\" rx=\"4\" ry=\"4\" width=\"117\" height=\"6.4\" />\n  \t<svg:rect x=\"70\" y=\"35\" rx=\"3\" ry=\"3\" width=\"85\" height=\"6.4\" />\n  \t<svg:rect x=\"0\" y=\"80\" rx=\"3\" ry=\"3\" width=\"350\" height=\"6.4\" />\n  \t<svg:rect x=\"0\" y=\"100\" rx=\"3\" ry=\"3\" width=\"380\" height=\"6.4\" />\n  \t<svg:rect x=\"0\" y=\"120\" rx=\"3\" ry=\"3\" width=\"201\" height=\"6.4\" />\n  \t<svg:circle cx=\"30\" cy=\"30\" r=\"30\" />\n   </content-loader>\n  "
            },] },
];
var ListContentLoaderComponent = /** @class */ (function (_super) {
    __extends(ListContentLoaderComponent, _super);
    function ListContentLoaderComponent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return ListContentLoaderComponent;
}(ContentLoaderComponent));
ListContentLoaderComponent.decorators = [
    { type: Component, args: [{
                selector: 'list-content-loader',
                template: "\n  <content-loader [animate]=\"animate\" [width]=\"width\" [height]=\"height\" [speed]=\"speed\" [preserveAspectRatio]=\"preserveAspectRatio\"\n  [primaryColor]=\"primaryColor\" [secondaryColor]=\"secondaryColor\" [primaryOpacity]=\"primaryOpacity\" [secondaryColor]=\"secondaryColor\"\n  [uniqueKey]=\"uniqueKey\">\n  \t<svg:rect x=\"0\" y=\"0\" rx=\"3\" ry=\"3\" width=\"250\" height=\"10\" />\n  \t<svg:rect x=\"20\" y=\"20\" rx=\"3\" ry=\"3\" width=\"220\" height=\"10\" />\n  \t<svg:rect x=\"20\" y=\"40\" rx=\"3\" ry=\"3\" width=\"170\" height=\"10\" />\n  \t<svg:rect x=\"0\" y=\"60\" rx=\"3\" ry=\"3\" width=\"250\" height=\"10\" />\n  \t<svg:rect x=\"20\" y=\"80\" rx=\"3\" ry=\"3\" width=\"200\" height=\"10\" />\n  \t<svg:rect x=\"20\" y=\"100\" rx=\"3\" ry=\"3\" width=\"80\" height=\"10\" />\n   </content-loader>\n  "
            },] },
];
var ContentLoaderModule = /** @class */ (function () {
    function ContentLoaderModule() {
    }
    ContentLoaderModule.forRoot = function () {
        return {
            ngModule: ContentLoaderModule
        };
    };
    return ContentLoaderModule;
}());
ContentLoaderModule.decorators = [
    { type: NgModule, args: [{
                imports: [CommonModule],
                declarations: [
                    ContentLoaderComponent,
                    FacebookContentLoaderComponent,
                    BulletListContentLoaderComponent,
                    ListContentLoaderComponent
                ],
                exports: [
                    ContentLoaderComponent,
                    FacebookContentLoaderComponent,
                    BulletListContentLoaderComponent,
                    ListContentLoaderComponent
                ]
            },] },
];

export { ContentLoaderComponent, ContentLoaderModule, FacebookContentLoaderComponent, ListContentLoaderComponent, BulletListContentLoaderComponent };
