// @eui/ux-commons/dist/bundles/ux-commons.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@eui/core')) :
	typeof define === 'function' && define.amd ? define(['exports', '@eui/core'], factory) :
	(factory((global.uxCommons = {}),global.core));
}(this, (function (exports,core) { 'use strict';

	/**
	 * @fileoverview added by tsickle
	 * @suppress {checkTypes} checked by tsc
	 */
	/* tslint:enable */

	/**
	 * @fileoverview added by tsickle
	 * @suppress {checkTypes} checked by tsc
	 */

	exports.UxOrderByPipe = core.UxOrderByPipe;
	exports.UxOrderByPipeModule = core.UxOrderByPipeModule;
	exports.UxFilterPipe = core.UxFilterPipe;
	exports.UxFilterPipeModule = core.UxFilterPipeModule;
	exports.UxTruncatePipe = core.UxTruncatePipe;
	exports.UxTruncatePipeModule = core.UxTruncatePipeModule;
	exports.UxCurrencyPipe = core.UxCurrencyPipe;
	exports.UxCurrencyPipeModule = core.UxCurrencyPipeModule;
	exports.UxNumberFormatPipe = core.UxNumberFormatPipe;
	exports.UxNumberFormatPipeModule = core.UxNumberFormatPipeModule;
	exports.UxMomentTzPipe = core.UxMomentTzPipe;
	exports.UxMomentTzPipeModule = core.UxMomentTzPipeModule;
	exports.UxIsoToCountryPipe = core.UxIsoToCountryPipe;
	exports.UxIsoToCountryPipeModule = core.UxIsoToCountryPipeModule;
	exports.UxScrollHandlerDirective = core.UxScrollHandlerDirective;
	exports.UxScrollHandlerElementDirective = core.UxScrollHandlerElementDirective;
	exports.UxScrollHandlerDirectiveModule = core.UxScrollHandlerDirectiveModule;
	exports.UxMaxlengthDirective = core.UxMaxlengthDirective;
	exports.UxMaxlengthModule = core.UxMaxlengthModule;
	exports.UxTooltip = core.UxTooltip;
	exports.UxTooltipModule = core.UxTooltipModule;
	exports.UxAutoResizeDirective = core.UxAutoResizeDirective;
	exports.UxAutoResizeModule = core.UxAutoResizeModule;
	exports.UxCollapsibleDirective = core.UxCollapsibleDirective;
	exports.UxCollapsibleDirectiveModule = core.UxCollapsibleDirectiveModule;
	exports.UxHasPermissionDirective = core.UxHasPermissionDirective;
	exports.UxHasPermissionDirectiveModule = core.UxHasPermissionDirectiveModule;
	exports.UxSmoothScrollToDirective = core.UxSmoothScrollToDirective;
	exports.UxSmoothScrollDirective = core.UxSmoothScrollDirective;
	exports.UxSmoothScrollDirectiveModule = core.UxSmoothScrollDirectiveModule;
	exports.UxCleaveDirective = core.UxCleaveDirective;
	exports.UxCleaveModule = core.UxCleaveModule;
	exports.NguiAutoCompleteDropdownButtonDirective = core.NguiAutoCompleteDropdownButtonDirective;
	exports.NguiAutoCompleteDropdownButtonModule = core.NguiAutoCompleteDropdownButtonModule;
	exports.UxTemplateDirective = core.UxTemplateDirective;
	exports.UxTemplateDirectiveModule = core.UxTemplateDirectiveModule;
	exports.UxLayoutAppShellComponent = core.UxLayoutAppShellComponent;
	exports.UxAppShellSidebarContentTagDirective = core.UxAppShellSidebarContentTagDirective;
	exports.UxAppShellSidebarHeaderContentTagDirective = core.UxAppShellSidebarHeaderContentTagDirective;
	exports.UxAppShellHeaderRightContentTagDirective = core.UxAppShellHeaderRightContentTagDirective;
	exports.UxAppShellHeaderTitleContentTagDirective = core.UxAppShellHeaderTitleContentTagDirective;
	exports.UxAppShellMainContentTagDirective = core.UxAppShellMainContentTagDirective;
	exports.UxAppShellNavBarItemsContentTagDirective = core.UxAppShellNavBarItemsContentTagDirective;
	exports.UxAppShellNavBarLeftItemsContentTagDirective = core.UxAppShellNavBarLeftItemsContentTagDirective;
	exports.UxAppShellFooterContentTagDirective = core.UxAppShellFooterContentTagDirective;
	exports.UxAppShellOverlayPanelHeaderTagDirective = core.UxAppShellOverlayPanelHeaderTagDirective;
	exports.UxAppShellTopMessageContentTagDirective = core.UxAppShellTopMessageContentTagDirective;
	exports.UxAppShellErrorStateContentTagDirective = core.UxAppShellErrorStateContentTagDirective;
	exports.UxLayoutAppShellComponentModule = core.UxLayoutAppShellComponentModule;
	exports.UxLayoutHeaderProfileComponent = core.UxLayoutHeaderProfileComponent;
	exports.UxLayoutHeaderProfileMenuContentTagDirective = core.UxLayoutHeaderProfileMenuContentTagDirective;
	exports.UxLayoutHeaderProfileNotLoggedInContentTagDirective = core.UxLayoutHeaderProfileNotLoggedInContentTagDirective;
	exports.UxLayoutHeaderProfileSubContentTagDirective = core.UxLayoutHeaderProfileSubContentTagDirective;
	exports.UxLayoutHeaderProfileComponentModule = core.UxLayoutHeaderProfileComponentModule;
	exports.UxLayoutSidebarItemsComponent = core.UxLayoutSidebarItemsComponent;
	exports.UxLayoutSidebarItemsComponentModule = core.UxLayoutSidebarItemsComponentModule;
	exports.UxLayoutSidebarItemComponent = core.UxLayoutSidebarItemComponent;
	exports.UxLayoutSidebarItemComponentModule = core.UxLayoutSidebarItemComponentModule;
	exports.UxLayoutNavBarActionItemComponent = core.UxLayoutNavBarActionItemComponent;
	exports.UxLayoutNavBarOverlayPanelContentTagDirective = core.UxLayoutNavBarOverlayPanelContentTagDirective;
	exports.UxLayoutNavBarItemContentTagDirective = core.UxLayoutNavBarItemContentTagDirective;
	exports.UxLayoutNavBarOverlayHeaderPanelContentTagDirective = core.UxLayoutNavBarOverlayHeaderPanelContentTagDirective;
	exports.UxLayoutNavBarActionItemComponentModule = core.UxLayoutNavBarActionItemComponentModule;
	exports.UxLayoutNavBarLeftActionItemComponent = core.UxLayoutNavBarLeftActionItemComponent;
	exports.UxLayoutNavBarLeftActionItemComponentModule = core.UxLayoutNavBarLeftActionItemComponentModule;
	exports.UxLayoutNavBarActionItemNotificationsComponent = core.UxLayoutNavBarActionItemNotificationsComponent;
	exports.UxLayoutNavBarActionItemNotificationsComponentModule = core.UxLayoutNavBarActionItemNotificationsComponentModule;
	exports.UxAppComponent = core.UxAppComponent;
	exports.UxAppModule = core.UxAppModule;
	exports.UxLayoutBreadcrumbsComponent = core.UxLayoutBreadcrumbsComponent;
	exports.UxLayoutBreadcrumbsComponentModule = core.UxLayoutBreadcrumbsComponentModule;
	exports.UxBreadcrumbsService = core.UxBreadcrumbsService;
	exports.UxLayoutOverlayPanelComponent = core.UxLayoutOverlayPanelComponent;
	exports.UxLayoutOverlayPanelComponentModule = core.UxLayoutOverlayPanelComponentModule;
	exports.UxLayoutTopMessageComponent = core.UxLayoutTopMessageComponent;
	exports.UxLayoutTopMessageComponentModule = core.UxLayoutTopMessageComponentModule;
	exports.UxLayoutBannerComponent = core.UxLayoutBannerComponent;
	exports.UxLayoutBannerComponentModule = core.UxLayoutBannerComponentModule;
	exports.UxLayoutHorizontalComponent = core.UxLayoutHorizontalComponent;
	exports.UxLayoutHorizontalLeftTagDirective = core.UxLayoutHorizontalLeftTagDirective;
	exports.UxLayoutHorizontalRightTagDirective = core.UxLayoutHorizontalRightTagDirective;
	exports.UxLayoutHorizontalCenterTagDirective = core.UxLayoutHorizontalCenterTagDirective;
	exports.UxLayoutHorizontalModule = core.UxLayoutHorizontalModule;
	exports.UxLayoutPageHeaderComponent = core.UxLayoutPageHeaderComponent;
	exports.UxLayoutPageHeaderActionsContentTagDirective = core.UxLayoutPageHeaderActionsContentTagDirective;
	exports.UxLayoutPageHeaderActionIconsContentTagDirective = core.UxLayoutPageHeaderActionIconsContentTagDirective;
	exports.UxLayoutPageHeaderComponentModule = core.UxLayoutPageHeaderComponentModule;
	exports.UxLayoutPageHeaderActionIconItemComponent = core.UxLayoutPageHeaderActionIconItemComponent;
	exports.UxLayoutPageHeaderActionIconItemComponentModule = core.UxLayoutPageHeaderActionIconItemComponentModule;
	exports.UxLayoutFooterActionBarComponent = core.UxLayoutFooterActionBarComponent;
	exports.UxLayoutFooterActionBarContentTagDirective = core.UxLayoutFooterActionBarContentTagDirective;
	exports.UxLayoutFooterActionBarComponentModule = core.UxLayoutFooterActionBarComponentModule;
	exports.UxLanguageSelectorComponent = core.UxLanguageSelectorComponent;
	exports.UxLanguageSelectorComponentModule = core.UxLanguageSelectorComponentModule;
	exports.UxLanguage = core.UxLanguage;
	exports.UxEuLanguages = core.UxEuLanguages;
	exports.UxSearchInputComponent = core.UxSearchInputComponent;
	exports.UxSearchInputComponentModule = core.UxSearchInputComponentModule;
	exports.UxTagComponent = core.UxTagComponent;
	exports.UxTagComponentModule = core.UxTagComponentModule;
	exports.UxIconComponent = core.UxIconComponent;
	exports.UxIconComponentModule = core.UxIconComponentModule;
	exports.UxIconToggleComponent = core.UxIconToggleComponent;
	exports.UxIconToggleComponentModule = core.UxIconToggleComponentModule;
	exports.UxIconBoxComponent = core.UxIconBoxComponent;
	exports.UxIconBoxComponentModule = core.UxIconBoxComponentModule;
	exports.UxMarkerComponent = core.UxMarkerComponent;
	exports.UxMarkerComponentModule = core.UxMarkerComponentModule;
	exports.UxLabelComponent = core.UxLabelComponent;
	exports.UxLabelContentTagDirective = core.UxLabelContentTagDirective;
	exports.UxLabelSubLabelTagDirective = core.UxLabelSubLabelTagDirective;
	exports.UxLabelComponentModule = core.UxLabelComponentModule;
	exports.UxDateTagComponent = core.UxDateTagComponent;
	exports.UxDateTagComponentModule = core.UxDateTagComponentModule;
	exports.UxToolbarFilterComponent = core.UxToolbarFilterComponent;
	exports.UxToolbarFilterButtonsTagDirective = core.UxToolbarFilterButtonsTagDirective;
	exports.UxToolbarFilterComponentModule = core.UxToolbarFilterComponentModule;
	exports.UxBlockDocumentComponent = core.UxBlockDocumentComponent;
	exports.UxBlockDocumentComponentModule = core.UxBlockDocumentComponentModule;
	exports.UxBlockContentComponent = core.UxBlockContentComponent;
	exports.UxBlockContentComponentModule = core.UxBlockContentComponentModule;
	exports.UxDisableContentComponent = core.UxDisableContentComponent;
	exports.UxDisableContentComponentModule = core.UxDisableContentComponentModule;
	exports.UxProgressCircleComponent = core.UxProgressCircleComponent;
	exports.UxProgressCircleComponentModule = core.UxProgressCircleComponentModule;
	exports.UxPanelComponent = core.UxPanelComponent;
	exports.UxPanelHeaderTagDirective = core.UxPanelHeaderTagDirective;
	exports.UxPanelFooterTagDirective = core.UxPanelFooterTagDirective;
	exports.UxPanelHeaderRightContentTagDirective = core.UxPanelHeaderRightContentTagDirective;
	exports.UxPanelHeaderStatusIndicatorContentTagDirective = core.UxPanelHeaderStatusIndicatorContentTagDirective;
	exports.UxPanelHeaderWithDescendantsTagDirective = core.UxPanelHeaderWithDescendantsTagDirective;
	exports.UxPanelFooterWithDescendantsTagDirective = core.UxPanelFooterWithDescendantsTagDirective;
	exports.UxPanelHeaderRightContentWithDescendantsTagDirective = core.UxPanelHeaderRightContentWithDescendantsTagDirective;
	exports.UxPanelSubHeaderContentTagDirective = core.UxPanelSubHeaderContentTagDirective;
	exports.UxPanelComponentModule = core.UxPanelComponentModule;
	exports.UxPanelsComponent = core.UxPanelsComponent;
	exports.UxPanelsComponentModule = core.UxPanelsComponentModule;
	exports.UxActionBoxComponent = core.UxActionBoxComponent;
	exports.UxActionBoxComponentModule = core.UxActionBoxComponentModule;
	exports.UxActionBoxItem = core.UxActionBoxItem;
	exports.UxAlertComponent = core.UxAlertComponent;
	exports.UxAlertComponentModule = core.UxAlertComponentModule;
	exports.UxDashboardCardComponent = core.UxDashboardCardComponent;
	exports.UxDashboardCardComponentModule = core.UxDashboardCardComponentModule;
	exports.UxControlFeedbackComponent = core.UxControlFeedbackComponent;
	exports.UxControlFeedbackComponentModule = core.UxControlFeedbackComponentModule;
	exports.UxFieldSearchComponent = core.UxFieldSearchComponent;
	exports.UxFieldSearchComponentModule = core.UxFieldSearchComponentModule;
	exports.UxFieldsetComponent = core.UxFieldsetComponent;
	exports.UxFieldsetLabelRightContentTagDirective = core.UxFieldsetLabelRightContentTagDirective;
	exports.UxFieldsetComponentModule = core.UxFieldsetComponentModule;
	exports.UxSwitchComponent = core.UxSwitchComponent;
	exports.UxSwitchComponentModule = core.UxSwitchComponentModule;
	exports.UxBadgeComponent = core.UxBadgeComponent;
	exports.UxBadgeComponentModule = core.UxBadgeComponentModule;
	exports.UxBadgeGroupComponent = core.UxBadgeGroupComponent;
	exports.UxBadgeGroupComponentModule = core.UxBadgeGroupComponentModule;
	exports.UxButtonComponent = core.UxButtonComponent;
	exports.UxButtonComponentModule = core.UxButtonComponentModule;
	exports.UxDropdownButtonComponent = core.UxDropdownButtonComponent;
	exports.UxDropdownButtonComponentModule = core.UxDropdownButtonComponentModule;
	exports.UxDropdownButtonItemComponent = core.UxDropdownButtonItemComponent;
	exports.UxDropdownButtonItemComponentModule = core.UxDropdownButtonItemComponentModule;
	exports.UxDropdownComponent = core.UxDropdownComponent;
	exports.UxDropdownComponentModule = core.UxDropdownComponentModule;
	exports.UxGrowlComponent = core.UxGrowlComponent;
	exports.UxGrowlComponentModule = core.UxGrowlComponentModule;
	exports.UxDashboardButtonsComponent = core.UxDashboardButtonsComponent;
	exports.UxDashboardButtonsComponentModule = core.UxDashboardButtonsComponentModule;
	exports.UxDashboardButtonComponent = core.UxDashboardButtonComponent;
	exports.UxDashboardButtonComponentModule = core.UxDashboardButtonComponentModule;
	exports.UxTabComponent = core.UxTabComponent;
	exports.UxTabComponentModule = core.UxTabComponentModule;
	exports.UxTabsComponent = core.UxTabsComponent;
	exports.UxTabsRightContentTagDirective = core.UxTabsRightContentTagDirective;
	exports.UxTabsComponentModule = core.UxTabsComponentModule;
	exports.UxTab = core.UxTab;
	exports.UxTabService = core.UxTabService;
	exports.UxTabsBarComponent = core.UxTabsBarComponent;
	exports.UxTabsBarRightContentTagDirective = core.UxTabsBarRightContentTagDirective;
	exports.UxTabsBarComponentModule = core.UxTabsBarComponentModule;
	exports.UxStaticTabsComponent = core.UxStaticTabsComponent;
	exports.UxStaticTabsComponentModule = core.UxStaticTabsComponentModule;
	exports.UxTimelineItemsComponent = core.UxTimelineItemsComponent;
	exports.UxTimelineItemsComponentModule = core.UxTimelineItemsComponentModule;
	exports.UxTimelineItemComponent = core.UxTimelineItemComponent;
	exports.UxTimelineItemComponentModule = core.UxTimelineItemComponentModule;
	exports.UxWizardStepComponent = core.UxWizardStepComponent;
	exports.UxWizardStepContentTagDirective = core.UxWizardStepContentTagDirective;
	exports.UxWizardStepComponentModule = core.UxWizardStepComponentModule;
	exports.UxWizardStepsComponent = core.UxWizardStepsComponent;
	exports.UxWizardStepsComponentModule = core.UxWizardStepsComponentModule;
	exports.UxWizardStep = core.UxWizardStep;
	exports.UxWizardStepService = core.UxWizardStepService;
	exports.UxTreeListComponent = core.UxTreeListComponent;
	exports.UxTreeListToolbarContentTagDirective = core.UxTreeListToolbarContentTagDirective;
	exports.UxTreeListComponentModule = core.UxTreeListComponentModule;
	exports.UxTreeListItemComponent = core.UxTreeListItemComponent;
	exports.UxTreeListItemSubLabelTagDirective = core.UxTreeListItemSubLabelTagDirective;
	exports.UxTreeListItemCustomContentTagDirective = core.UxTreeListItemCustomContentTagDirective;
	exports.UxTreeListItemRightContentTagDirective = core.UxTreeListItemRightContentTagDirective;
	exports.UxTreeListItemDetailsContentTagDirective = core.UxTreeListItemDetailsContentTagDirective;
	exports.UxTreeListItemSubContainerContentTagDirective = core.UxTreeListItemSubContainerContentTagDirective;
	exports.UxTreeListItemComponentModule = core.UxTreeListItemComponentModule;
	exports.UxModalComponent = core.UxModalComponent;
	exports.UxModalBodyTagDirective = core.UxModalBodyTagDirective;
	exports.UxModalFooterTagDirective = core.UxModalFooterTagDirective;
	exports.UxModalHeaderTagDirective = core.UxModalHeaderTagDirective;
	exports.UxModalComponentModule = core.UxModalComponentModule;
	exports.UxCarouselItemComponent = core.UxCarouselItemComponent;
	exports.UxCarouselItemComponentModule = core.UxCarouselItemComponentModule;
	exports.UxCarouselItemsComponent = core.UxCarouselItemsComponent;
	exports.UxCarouselItemsComponentModule = core.UxCarouselItemsComponentModule;
	exports.UxTreeComponent = core.UxTreeComponent;
	exports.UxTreeNodeComponent = core.UxTreeNodeComponent;
	exports.UxTreeComponentModule = core.UxTreeComponentModule;
	exports.UxTimebarComponent = core.UxTimebarComponent;
	exports.UxTimebarComponentModule = core.UxTimebarComponentModule;
	exports.UxTimebarItem = core.UxTimebarItem;
	exports.UxAutocompleteTagItem = core.UxAutocompleteTagItem;
	exports.UxAutocompleteTagComponent = core.UxAutocompleteTagComponent;
	exports.UxAutocompleteTagComponentModule = core.UxAutocompleteTagComponentModule;
	exports.UxButtonGroupComponent = core.UxButtonGroupComponent;
	exports.UxButtonGroupComponentModule = core.UxButtonGroupComponentModule;
	exports.UxButtonGroupItemComponent = core.UxButtonGroupItemComponent;
	exports.UxButtonGroupItemComponentModule = core.UxButtonGroupItemComponentModule;
	exports.UxSplitButtonComponent = core.UxSplitButtonComponent;
	exports.UxSplitButtonComponentModule = core.UxSplitButtonComponentModule;
	exports.UxMessageBoxComponent = core.UxMessageBoxComponent;
	exports.UxMessageBoxFooterTagDirective = core.UxMessageBoxFooterTagDirective;
	exports.UxMessageBoxComponentModule = core.UxMessageBoxComponentModule;
	exports.UxOverlayComponent = core.UxOverlayComponent;
	exports.UxOverlayComponentModule = core.UxOverlayComponentModule;
	exports.UxNotificationItemComponent = core.UxNotificationItemComponent;
	exports.UxNotificationItemComponentModule = core.UxNotificationItemComponentModule;
	exports.UxNotificationsPanelComponent = core.UxNotificationsPanelComponent;
	exports.UxNotificationsPanelComponentModule = core.UxNotificationsPanelComponentModule;
	exports.UxStickyComponent = core.UxStickyComponent;
	exports.UxStickyComponentModule = core.UxStickyComponentModule;
	exports.UxPieChartComponent = core.UxPieChartComponent;
	exports.UxPieChartCustomLegendTagDirective = core.UxPieChartCustomLegendTagDirective;
	exports.UxPieChartComponentModule = core.UxPieChartComponentModule;
	exports.UxPieChartLegendItemComponent = core.UxPieChartLegendItemComponent;
	exports.UxPieChartLegendItemComponentModule = core.UxPieChartLegendItemComponentModule;
	exports.UxPieChartGroupComponent = core.UxPieChartGroupComponent;
	exports.UxPieChartGroupComponentModule = core.UxPieChartGroupComponentModule;
	exports.UxDynamicMessageBoxService = core.UxDynamicMessageBoxService;
	exports.UxDynamicMessageBoxComponent = core.UxDynamicMessageBoxComponent;
	exports.UxDynamicMessageBoxComponentModule = core.UxDynamicMessageBoxComponentModule;
	exports.UxDynamicMessageBoxConfig = core.UxDynamicMessageBoxConfig;
	exports.UxDynamicModalComponent = core.UxDynamicModalComponent;
	exports.UxDynamicModalBodyTagDirective = core.UxDynamicModalBodyTagDirective;
	exports.UxDynamicModalFooterTagDirective = core.UxDynamicModalFooterTagDirective;
	exports.UxDynamicModalHeaderTagDirective = core.UxDynamicModalHeaderTagDirective;
	exports.UxDynamicMessageBoxFooterTagDirective = core.UxDynamicMessageBoxFooterTagDirective;
	exports.UxDynamicModalComponentModule = core.UxDynamicModalComponentModule;
	exports.UxDynamicModalService = core.UxDynamicModalService;
	exports.UxItemComponent = core.UxItemComponent;
	exports.UxItemComponentModule = core.UxItemComponentModule;
	exports.UxListItemComponent = core.UxListItemComponent;
	exports.UxListItemSubLabelTagDirective = core.UxListItemSubLabelTagDirective;
	exports.UxListtemContentTagDirective = core.UxListtemContentTagDirective;
	exports.UxListItemComponentModule = core.UxListItemComponentModule;
	exports.UxListItemsComponent = core.UxListItemsComponent;
	exports.UxListItemsComponentModule = core.UxListItemsComponentModule;
	exports.UxMenuItemComponent = core.UxMenuItemComponent;
	exports.UxMenuItemComponentModule = core.UxMenuItemComponentModule;
	exports.UxLoadingIndicatorComponent = core.UxLoadingIndicatorComponent;
	exports.UxLoadingIndicatorComponentModule = core.UxLoadingIndicatorComponentModule;
	exports.UxPopoverComponent = core.UxPopoverComponent;
	exports.UxPopoverComponentModule = core.UxPopoverComponentModule;
	exports.UxDiscussionThreadComponent = core.UxDiscussionThreadComponent;
	exports.UxDiscussionThreadComponentModule = core.UxDiscussionThreadComponentModule;
	exports.UxDiscussionThreadItemComponent = core.UxDiscussionThreadItemComponent;
	exports.UxDiscussionThreadItemComponentModule = core.UxDiscussionThreadItemComponentModule;
	exports.UxDiscussionThreadItem = core.UxDiscussionThreadItem;
	exports.UxFormControlComponent = core.UxFormControlComponent;
	exports.UxFormControlComponentModule = core.UxFormControlComponentModule;
	exports.UxFormGroupComponent = core.UxFormGroupComponent;
	exports.UxFormGroupSubContentTagDirective = core.UxFormGroupSubContentTagDirective;
	exports.UxFormGroupComponentModule = core.UxFormGroupComponentModule;
	exports.UxDynamicPopoverService = core.UxDynamicPopoverService;
	exports.UxDynamicPopoverConfig = core.UxDynamicPopoverConfig;
	exports.UxFilterResultsComponent = core.UxFilterResultsComponent;
	exports.UxFilterResultsNavigationContentTagDirective = core.UxFilterResultsNavigationContentTagDirective;
	exports.UxFilterResultsFilterContentTagDirective = core.UxFilterResultsFilterContentTagDirective;
	exports.UxFilterResultsContentTagDirective = core.UxFilterResultsContentTagDirective;
	exports.UxFilterResultsQuickFilterContentTagDirective = core.UxFilterResultsQuickFilterContentTagDirective;
	exports.UxFilterResultsHeaderContentTagDirective = core.UxFilterResultsHeaderContentTagDirective;
	exports.UxFilterResultsFullContentTagDirective = core.UxFilterResultsFullContentTagDirective;
	exports.UxFilterResultsComponentModule = core.UxFilterResultsComponentModule;
	exports.UxEditorComponent = core.UxEditorComponent;
	exports.UxEditorComponentModule = core.UxEditorComponentModule;
	exports.UxMessage = core.UxMessage;
	exports.UxButtonConfig = core.UxButtonConfig;
	exports.UxDynamicModalConfig = core.UxDynamicModalConfig;
	exports.UxModule = core.UxModule;
	exports.UxAllModule = core.UxAllModule;
	exports.UX_COLORS = core.UX_COLORS;
	exports.UX_COUNTRIES = core.UX_COUNTRIES;
	exports.UX_I18N = core.UX_I18N;
	exports.UX_TIMEZONES = core.UX_TIMEZONES;
	exports.UxLink = core.UxLink;
	exports.UxBadge = core.UxBadge;
	exports.UxService = core.UxService;
	exports.UxAppShellService = core.UxAppShellService;
	exports.UxTimezoneService = core.UxTimezoneService;
	exports.UxDynamicComponentService = core.UxDynamicComponentService;
	exports.UxPermissionService = core.UxPermissionService;
	exports.StorageService = core.StorageService;
	exports.UxDomService = core.UxDomService;
	exports.UxAssetsService = core.UxAssetsService;
	exports.DYNAMIC_COMPONENT_CONFIG = core.DYNAMIC_COMPONENT_CONFIG;
	exports.CONFIG_TOKEN = core.CONFIG_TOKEN;

	Object.defineProperty(exports, '__esModule', { value: true });

})));
